### PR TITLE
Clear SDFGI textures when created

### LINKED ISSUE #77427: Enabling SDFGI for the first time causes artefacts due to corrupted textures
### Godot version

v4.0.3.stable.official

### System information

Windows 10, Vulkan (Forward+), GTX 1060

### Issue description

When turning on SDFGI for the first time, you will see highly bright lights or sometimes random colours. Sometimes it does not happen and seems to be very random.

https://github.com/godotengine/godot/assets/76781125/029d1e8b-7bc7-43b0-883b-a88fefb19548

I used Renderdoc, and I may have found the issue, though I have very little idea how to actually use this. I found 5 textures that were corrupted and all look like this:
image

they are all labelled as VoxelGI though I think this is some mistake as my scene doesn't use voxelgi. Regardless, these textures seem to be the cause of the strange artefacts, because the issue seems to go away after some frames, and the textures return to normal.

these textures are:
lightprobe_history_texture
lightprobe_average_texture
lightprobe_history_scroll_texture
lightprobe_average_scroll_texture
lightprobe_average_parent_texture

### Steps to reproduce

Enable SDFGI for the first time in a scene, and you may see how the colours start off extremely bright and random, then slowly calm down to form a normal image.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by bitsawer**
* Fixes https://github.com/godotengine/godot/issues/77427

Related discussion: https://github.com/godotengine/godot/issues/39961#issuecomment-966686804

Note that visual artifacts also happens when changing some  SDFGI parameters, not just when it's first created.

Clears all SDFGI textures when they are created to avoid visual artifacts from uninitialized data. This is a shotgun approach to clear everything, I tried to find individual textures responsible for that glitching, but I couldn't isolate any single confirmed culprit. The existing code does clear a few textures already, but as the bug report shows it doesn't seem to be enough. Someone more familiar with this code might figure out if there is a more limited set of textures that have to be cleared.

The issue itself is pretty easy to repro even with a simple scene, toggling environment SDFGI on and off and adjusting its parameters results in those visual glitches pretty often. Resizing the window and tweaking the parameter also seems to also show the issue quickly. After this PR, I cant't repro the issue anymore.

Also changes texture debug names from VoxelGI to SDFGI, looks like a mistake to me (and also to the person who wrote the bug report).

### CODE REVIEW

**Review by Calinou — APPROVED** [review]
Tested locally (Linux + NVIDIA), it works as expected.

In a future PR, I wonder if we initialize SDFGI color to approximately match the environment color when **Read Sky Light** is enabled. This makes the transition when toggling SDFGI less jarring, which is nice to have when initially loading a scene (so you don't have to hide the transition as much).

**Comment by jcostello** [comment]
> Tested locally (Linux + NVIDIA), it works as expected.
> 
> In a future PR, I wonder if we initialize SDFGI color to approximately match the environment color when **Read Sky Light** is enabled. This makes the transition when toggling SDFGI less jarring, which is nice to have when initially loading a scene (so you don't have to hide the transition as much).

A signal when finish could fix when loading a scene

**Comment by bitsawer** [comment]
It would be interesting to test the skylight color tuning or even instead of starting from 0 initialized values try to converge from old values. But the latter is that's kind of https://github.com/godotengine/godot/issues/39961 territory.

Not sure if SDFGI ready signal is needed, I assume that you can just wait Project Settings **Frames To Converge** amount of frames and it should be ready. Not sure though, haven't tested it.

However, one useful thing would be to force converge the SDFGI when loading a scene instead of sitting and waiting until enough frames have gone by. A basic experiment with `RenderingServer.force_draw()` like this doesn't seem to work, but maybe there is some other way;

```gdscript
node.environment.sdfgi_enabled = true

for i in range(30): #Or whatever your SDFGI convergence setting is
    RenderingServer.force_sync()
    RenderingServer.force_draw(true, 0.1)
```

File: servers/rendering/renderer_rd/environment/gi.cpp [hunk 1]
```diff
@@ -543,25 +540,19 @@ void GI::SDFGI::create(RID p_env, const Vector3 &p_world_position, uint32_t p_re
...
-		RD::get_singleton()->set_resource_name(cascade.light_aniso_1_tex, "VoxelGI Cascade Light Aniso 1 Texture");
+		cascade.light_aniso_0_tex = create_clear_texture(tf_aniso0, "SDFGI Cascade Light Aniso 0 Texture");
+		cascade.light_aniso_1_tex = create_clear_texture(tf_aniso1, "SDFGI Cascade Light Aniso 1 Texture");
 
 		{
 			RD::TextureView tv;
 			tv.format_override = RD::DATA_FORMAT_E5B9G9R9_UFLOAT_PACK32;
 			cascade.light_tex = RD::get_singleton()->texture_create_shared(tv, cascade.light_data);
 
 			RD::get_singleton()->texture_clear(cascade.light_tex, Color(0, 0, 0, 0), 0, 1, 0, 1);
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/environment/gi.cpp:555]: I guess we don't need to clear this since ``light_data`` is now cleared on creation

**Review by clayjohn — COMMENTED** [review]
Looks good to me, just left one little comment

File: servers/rendering/renderer_rd/environment/gi.cpp (hunk 1 above)
Comment (bitsawer) [reply servers/rendering/renderer_rd/environment/gi.cpp]: Fixed, I forgot to remove the shared texture clear when I started digging to see if it needed any special handling and then forgot to actually remove it,

**Review by clayjohn — APPROVED** [review]
Looks good to me. I also tested locally and this appears to work well

**Comment by akien-mga** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.1.2.

**Pull request merged**